### PR TITLE
Upgrade go (1.19) and libp2p (v0.24)

### PR DESCRIPTION
**Pull request by amir-blox**
- Updated Go `1.17` > `1.19`
- Update libp2p (`v0.24.2`)
  - updated package imports
  - fix breaking changes
  - fix mdns discovery
- conn handler: avoid disconnecting outbound connections
- discovery flag `p2p.Discovery` or `P2P_DISCOVERY`
- updated docker files and guides

### CODE REVIEW

File: network/p2p/config.go [hunk 1]
```diff
@@ -21,13 +21,15 @@ import (
...
 const (
+	localDiscvery  = "mdns"
 	minPeersBuffer = 10
 )
 
 // Config holds the configuration options for p2p network
 type Config struct {
 	// prod enr
 	Bootnodes string `yaml:"Bootnodes" env:"BOOTNODES" env-description:"Bootnodes to use to start discovery, seperated with ';'" env-default:"enr:-Li4QO2k62g1tiwitaoFVMT8zN-sSNPp8cg8Kv-5lg6_6VLjVZREhxVMSmerOTptlKbBaO2iszi7rvKBYzbGf38HpcSGAYLoed50h2F0dG5ldHOIAAAAAAAAAACEZXRoMpD1pf1CAAAAAP__________gmlkgnY0gmlwhCLdWuKJc2VjcDI1NmsxoQITQ1OchoBl5XW9RfBembdN9Er1qNEOIc5ohrQ0rT9B-YN0Y3CCE4iDdWRwgg-g;enr:-Li4QAxqhjjQN2zMAAEtOF5wlcr2SFnPKINvvlwMXztJhClrfRYLrqNy2a_dMUwDPKcvM7bebq3uptRoGSV0LpYEJuyGAYRZG5n5h2F0dG5ldHOIAAAAAAAAAACEZXRoMpD1pf1CAAAAAP__________gmlkgnY0gmlwhBLb3g2Jc2VjcDI1NmsxoQLbXMJi_Pq3imTq11EwH8MbxmXlHYvH2Drz_rsqP1rNyoN0Y3CCE4iDdWRwgg-g"`
+	Discovery string `yaml:"Discovery" env:"P2P_DISCOVERY" env-description:"Discovery system to use" env-default:"discv5"`
```
Comment (nivBlox) [inline network/p2p/config.go:32]: this flag is for running mdns?

Comment (nivBlox) [reply network/p2p/config.go:32]: if so, let's add the flag options in the description to ease

**Review by moshe-blox — APPROVED** [review]
🔥

File: network/p2p/config.go (hunk 1 above)
Comment (amir-blox) [reply network/p2p/config.go:32]: Yes

**Pull request merged**